src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMutation.cs

```diff
@@ -0,0 +1,279 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    // Patch(dataSource:*[], Record, Updates1, Updates2,…)
+    internal class PatchImpl : PatchFunction, IAsyncTexlFunction5
+    {
+        public async Task<FormulaValue> InvokeAsync(IServiceProvider runtimeServiceProvider, FormulaType irContext, FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            MutationProcessUtils.ThrowIfPFxV1NotActive(runtimeServiceProvider.GetService<Features>(), "Patch");
+
+            var arg0 = args[0];
+
+            if (args[0] is LambdaFormulaValue arg0lazy)
+            {
+                arg0 = await arg0lazy.EvalAsync().ConfigureAwait(false);
+            }
+
+            if (arg0 is not TableValue tableValue)
+            {
+                return arg0;
+            }
+
+            if (args[1] is not RecordValue baseRecord)
+            {
+                return args[1];
+            }
+
+            var dvalue = MutationProcessUtils.MergeRecords(args.Skip(2));
+
+            if (!dvalue.IsValue)
+            {
+                return dvalue.ToFormulaValue();
+            }
+
+            // If base record is {}, then collect.
```

Comment:
>// If base record is {}, then collect. [](http://example.com/codeflow?start=12&length=38)

Not exactly; we will collect if we cannot find the base record in the table. For example, after `ClearCollect(coll, {a:1,b:2},{a:3,b:4}); Patch(coll, {a:2,b:3}, {a:5,b:6})`, coll will have 3 elements - `[{"a":1,"b":2},{"a":3,"b":4},{"a":5,"b":6}]`

Reply:
Yes, this is the topic that I brought to discussion internally. Once I have an answer I'll address this.

Reply:
If the record is found in the table, update it. Otherwise, create it. For now, we are matching all fields to find the record. This is still open to discussion.